components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMGroupResolver.java

```diff
@@ -528,11 +528,26 @@ public boolean listGroups(Condition condition, int limit, int offset, String dom
         }
         ExpressionCondition expressionCondition = (ExpressionCondition) condition;
         String attributeName = resolveGroupAttributeWithSCIMSchema(expressionCondition.getAttributeName(), tenantId);
-        String attributeValue = buildSearchAttributeValue(attributeName, expressionCondition.getOperation(),
-                expressionCondition.getAttributeValue(), SQL_FILTERING_DELIMITER);
+        String attributeValue;
+        if (StringUtils.equals(SCIMConstants.GroupSchemaConstants.DISPLAY_NAME_URI, attributeName)) {
+            String nameWithDomain = expressionCondition.getAttributeValue();
+            if (!nameWithDomain.contains(CarbonConstants.DOMAIN_SEPARATOR)) {
+                nameWithDomain = domain + CarbonConstants.DOMAIN_SEPARATOR + nameWithDomain;
```

Comment:
Can `domain` be null?

Reply:
In the application mgt side I have a logic to fallback this if the domain is null.
<img width="486" alt="image" src="https://github.com/user-attachments/assets/80c9f502-da6c-4a11-999a-14beccd7f4a1" />

However, since this may be used in other places in the future, I will add a null check here and assign the PRIMARY domain.

Reply:
Addressed by https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/619/commits/36a9f98f01cf4bbd8590c2cebf7d7b7d54234c8a